k2/csrc/log.h

```diff
@@ -134,6 +136,23 @@ class Voidifier {
 
 }  // namespace k2
 
+/**
+ * @brief A macro to get checked at compile time, as static assertion
+ *
+ * @param[in] COND the compile-time boolean expression that must be true
+ * @param[in] MSG  an error message if exp is false
+ *
+ * @code{.cpp}
+ * K2_STATIC_ASSERT(DEFINED_SHAPE % DEFINED_X == 0);
+ * @endcode
+ */
+#define K2_STATIC_ASSERT4(COND, MSG) \
```

Comment:
IIRC, we have decided to use the version `static_assert` in  `C++11`. (And define some macro like `#define K2_STATIC_ASSERT(x) static_assert(x, "")`)

Reply:
changed it as suggested